sonarcloud/resource_project_main_branch.go

```diff
@@ -0,0 +1,189 @@
+package sonarcloud
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/reinoudk/go-sonarcloud/sonarcloud/project_branches"
+)
+
+type resourceProjectMainBranchType struct{}
+
+func (r resourceProjectMainBranchType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Description: `This resource manages a project main branch.
+
+Note that certain operations, such as the deletion of a project's main branch configuration, may
+not be permitted by the SonarCloud web API, or may require admin permissions.
+		`,
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				Type:     types.StringType,
+				Computed: true,
+			},
+			"name": {
+				Type:        types.StringType,
+				Required:    true,
+				Description: "The name of the project main branch.",
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					tfsdk.RequiresReplace(),
+				},
+				Validators: []tfsdk.AttributeValidator{
+					stringLengthBetween(1, 255),
+				},
+			},
+			"project_key": {
+				Type:        types.StringType,
+				Required:    true,
+				Description: "The key of the project.",
+				Validators: []tfsdk.AttributeValidator{
+					stringLengthBetween(1, 400),
+				},
+			},
+		},
+	}, nil
+}
+
+func (r resourceProjectMainBranchType) NewResource(_ context.Context, p tfsdk.Provider) (tfsdk.Resource, diag.Diagnostics) {
+	return resourceProject{
```

Comment:
Small copy-paste leftover ;)

```suggestion
	return resourceProjectMainBranch{
```